### ISSUE TITLE
Frequent pilot crashes in CI after enabling analysis by default

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#27978]
https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/27963/integ-security-k8s-tests_istio/21821/artifacts/security-filebased-tls-origination-/_suite_context/istio-state-855522116/cluster-0/istiod-745b5f7c8b-tpgrz_discovery.previous.log
```
fatal error: concurrent map writes
fatal error: concurrent map writes
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x253b873]

goroutine 1766 [running]:
istio.io/istio/pilot/pkg/status.(*DistributionController).writeStatus(0xc001a8c120, 0x3236ae0, 0xc000aadb40, 0xc003600120, 0x13, 0xc003600134, 0x8, 0xc00360013d, 0xc, 0xc00360014a, ...)
	istio.io/istio/pilot/pkg/status/state.go:176 +0x193
created by istio.io/istio/pilot/pkg/status.(*DistributionController).writeAllStatus
	istio.io/istio/pilot/pkg/status/state.go:154 +0x519

```

cc @stevenctl @nmittler

**Comment by howardjohn** [comment]
not stale

**Comment by therealmitchconnors** [comment]
I believe this is fixed by #29275.

**Comment by therealmitchconnors** [comment]
This should be fixed now.  https://github.com/istio/istio/pull/30187 re-enables this feature for tests.
